sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobApiTests.java

```diff
@@ -2891,27 +2891,43 @@ public void getBlobName(String inputName, String expectedOutputName) {
 
     private static Stream<Arguments> getBlobNameSupplier() {
         return Stream.of(
-            Arguments.of("blobName", "blobName"), // standard names should be preserved
-            // encoded names should be decoded (not double decoded))
-            Arguments.of(Utility.urlEncode("dir1/a%20b.txt"), "dir1/a%20b.txt"));
+            Arguments.of("blobName", "blobName"),
+            Arguments.of("dir1/a%20b.txt", "dir1/a%20b.txt"),
+            Arguments.of("path/to]a blob", "path/to]a blob"),
+            Arguments.of("path%2Fto%5Da%20blob", "path%2Fto%5Da%20blob"),
+            Arguments.of("斑點", "斑點"),
+            Arguments.of("%E6%96%91%E9%BB%9E", "%E6%96%91%E9%BB%9E"),
+            Arguments.of("斑點", "斑點"));
     }
 
     @ParameterizedTest
-    @MethodSource("getBlobNameAndBuildClientSupplier")
+    @MethodSource("getBlobNameSupplier")
     public void getBlobNameAndBuildClient(String originalBlobName, String finalBlobName) {
         BlobClient client = cc.getBlobClient(originalBlobName);
         BlobClientBase baseClient = cc.getBlobClient(client.getBlobName()).getBlockBlobClient();
 
         assertEquals(baseClient.getBlobName(), finalBlobName);
     }
 
-    private static Stream<Arguments> getBlobNameAndBuildClientSupplier() {
+    private static Stream<Arguments> getNonEncodedBlobNameSupplier() {
         return Stream.of(
-            Arguments.of("blob", "blob"),
-            Arguments.of("path/to]a blob", "path/to]a blob"),
-            Arguments.of("path%2Fto%5Da%20blob", "path/to]a blob"),
-            Arguments.of("斑點", "斑點"),
-            Arguments.of("%E6%96%91%E9%BB%9E", "斑點"));
+            Arguments.of("test%test"),
+            Arguments.of("ab2a7d5f-b973-4222-83ba-d0581817a819 %Россия 한국 中国!?/file"),
+            Arguments.of("%E6%96%91%E9%BB%9E"),
+            Arguments.of("斑點"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("getNonEncodedBlobNameSupplier")
+    public void getNonEncodedBlobName(String originalBlobName) {
+        BlobClient client = cc.getBlobClient(originalBlobName);
+        BlockBlobClient blockBlobClient = cc.getBlobClient(client.getBlobName()).getBlockBlobClient();
+        assertEquals(blockBlobClient.getBlobName(), originalBlobName);
+
+        // now try uploading to the portal and see if the blob name will be properly encoded in the url
+        blockBlobClient.upload(DATA.getDefaultInputStream(), DATA.getDefaultDataSize());
```

Comment:
I don't think we actually need to do an upload here, we just care about the assert on 2930.